### PR TITLE
[DLS] Add "last_access_control_sync_error" to protocol docs

### PR DESCRIPTION
**Pull request by timgrein**
## Related to https://github.com/elastic/enterprise-search-team/issues/4695

This PR adds the new field `last_access_control_sync_error` to the protocol docs.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

**Review by seanstory — DISMISSED** [review]

File: docs/CONNECTOR_PROTOCOL.md [hunk 1]
```diff
@@ -114,13 +114,14 @@ This is our main communication index, used to communicate the connector's config
   index_name: string;   -> The name of the content index where data will be written to
   is_native: boolean;   -> Whether this is a native connector
   language: string;     -> the language used for the analyzer
+  last_access_control_sync_error: keyword;   -> Optional last access control sync job error message
```
Comment (wangch079) [inline docs/CONNECTOR_PROTOCOL.md:117]:
```suggestion
  last_access_control_sync_error: string;   -> Optional last access control sync job error message
```

**Review by wangch079 — COMMENTED** [review]
🤔 Why did we choose `keyword` for `last_sync_error`?

File: docs/CONNECTOR_PROTOCOL.md (hunk 1 above)
Comment (timgrein) [reply docs/CONNECTOR_PROTOCOL.md:117]: Enterprise Search actually uses `last_sync_error: { type: 'keyword' }` as mapping for `last_sync_error`, so I simply used the same type to be consistent. So I guess our docs are wrong or we need to change the mapping for `last_sync_error` and `last_access_control_sync_error`.

Comment (wangch079) [reply docs/CONNECTOR_PROTOCOL.md:117]: > Enterprise Search actually uses `last_sync_error: { type: 'keyword' }` as mapping for `last_sync_error`

Noticed this as well: https://github.com/elastic/connectors-python/pull/1011#pullrequestreview-1464065947

But the suggestion above is not the exact data type. `last_sync_error` also uses `string`.
https://github.com/elastic/connectors-python/blob/013b0ada6301ed92fef1cbab793d31c24e7a01c7/docs/CONNECTOR_PROTOCOL.md?plain=1#L123

**Pull request merged**